### PR TITLE
Add logic to allow services on ingress rules

### PR DESCRIPTION
**Pull request by mgleung**
## Description

Allows services to be specified in the Source portion of the Ingress rules of policies

## Todos
- [ ] Tests
- [ ] Documentation
- [ ] Release note

## Release Note

```release-note
Allow services to be specified in the Source field of Ingress rules
```

### CODE REVIEW

File: lib/backend/syncersv1/updateprocessors/rules_test.go [hunk 1]
```diff
@@ -450,7 +450,7 @@ var _ = Describe("Test the Rules Conversion Functions", func() {
 		})
 	})
 
-	It("should parse a services match", func() {
+	It("should parse a destination rule services match", func() {
```
Comment (caseydavenport) [inline lib/backend/syncersv1/updateprocessors/rules_test.go:453]: egress rule instead of destination rule

File: lib/validator/v3/validator.go [hunk 2]
```diff
@@ -1059,6 +1059,17 @@ func validateRule(structLevel validator.StructLevel) {
...
+
+	// Check that destination service rules do not use ports.
+	// Destination service rules use ports specified on the endpoints.
+	if rule.Destination.Services != nil && len(rule.Destination.Ports) != 0 {
+		structLevel.ReportError(reflect.ValueOf(rule.Destination.Ports),
+			"Destination.Ports", "", reason("cannot specify ports with a service selector"), "")
+	}
+	if rule.Destination.Services != nil && len(rule.Destination.NotPorts) != 0 {
+		structLevel.ReportError(reflect.ValueOf(rule.Destination.NotPorts),
+			"Destination.NotPorts", "", reason("cannot specify ports with a service selector"), "")
```
Comment (caseydavenport) [inline lib/validator/v3/validator.go:1071]: "cannot specify notports"

**Review by caseydavenport — APPROVED** [review]
A few minor comments but otherwise LGTM

File: lib/validator/v3/validator_test.go [hunk 3]
```diff
@@ -2141,6 +2158,50 @@ func init() {
...
+									Name:      "service1",
+									Namespace: "default",
+								},
+							},
+						},
+					},
+				},
+			}, true,
+		),
+		Entry("allow a Service match AND a NotPorts match on a source rule",
```
[the remark this answers is not in the document]
Comment (mgleung) [reply lib/validator/v3/validator_test.go:2183]: I think I have these cases covered by the following test cases:
* "allow a Service match in an egress rule destination"
* "disallow a Service match in an egress rule source"
* "allow a Service match in an ingress rule source"
* "disallow a Service match in an ingress rule destination"

**Pull request merged**